doc/api/fs.md

```diff
@@ -3572,8 +3572,15 @@ console.log(fs.readFileSync('temp.txt', 'utf8'));
 // Prints: Node.js
 
 async function doTruncate() {
-  const fd = await fsPromises.open('temp.txt', 'r+');
-  await fsPromises.ftruncate(fd, 4);
+  let filehandle;
+  try {
+    filehandle = await fsPromises.open('temp.txt', 'r+');
+    await filehandle.truncate(4);
+  } finally {
+    if (filehandle !== null) {
```

Comment:
What can make undefined `filehandle` be `null`?

Reply:
The `open` throwing, since that was unclear it'll definitely be unclear for some users - we should probably add a comment.

Reply:
If open throws, will not the `filehandle` remain `undefined`?

Reply:
I could have sworn that was a `!=` and not a `!==`, good spotting!

Reply:
I mean, shoud not we initialize `filehandle` as `null` or use non-strict comparison or even `if (filehandle)` ?

Reply:
We should. I was saying "good spotting" for spotting that.

Reply:
Sorry, GitHub did not update timely with new comments)

Reply:
@vsemozhetbyt @benjamingr 
`filehandle` is not `undefined`. It is defined and initialized as `null`. 
```js
let filehandle;
```

But it is unclear for some users. 
The below code may be more clearly for users. It looks better?

```js
async function doTruncate() {
  let filehandle = null;
  try {
    filehandle = await fsPromises.open('temp.txt', 'r+');
    await filehandle.truncate(4);
  } finally {
    if (filehandle) {
      // close the file if it is opened.
      await filehandle.close();
    }
  }
  console.log(fs.readFileSync('temp.txt', 'utf8'));  // Prints: Node
}
```

Reply:
```
let filehandle;
```

When you do that in JavaScript that initializes the value to `undefined`. You can verify this in the following script:

```js
let filehandle;
console.log(filehandle);
```

> The below code may be more clearly for users. It looks better?

Yes, thank you.

Reply:
@benjamingr 
oh, sorry! I was wrong and fixed them.